tests/vmi_multiqueue_test.go

```diff
@@ -154,3 +158,12 @@ var _ = Describe("[sig-compute]MultiQueue", func() {
 
 	})
 })
+
+func newFedoraVMIWithAutoattachedInterfaces() *v1.VirtualMachineInstance {
```

Comment:
- I do not see where "Autoattach" is set, is this implicit?
- How is `tests.NewRandomVMIWithEphemeralDisk` any different here?
- Why do you need Fedora & could-init here?

Reply:
> * I do not see where "Autoattach" is set, is this implicit?
AutoAttach is enable by default yes (ptr value == nil), or when the ptr value is not nil, and the *ptr is true

> * How is `tests.NewRandomVMIWithEphemeralDisk` any different here?
It adds an interface, and we need a machine with auto attached interfaces only

> * Why do you need Fedora & could-init here?
checking login past the selinux when MQ is enabled, (added a table in the current test),
will check about the cloud-init if needed

Reply:
@EdDev  [`AutoattachPodInterface` is true by default.](https://github.com/kubevirt/kubevirt/blob/a78d4b950a36c1df345f885e752247d3ed019eac/pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go#L206-L205). If the vm/vmi has no interfaces, and `AutoattachPodInterface` is not set at all and set to `true` a default pod interface will be added.

> The ref BZ suggested to default the behavior to remove the multique implicitly when it is not valid, and not the throw an error.
> 
> > if there are no interfaces just don't set the number of queues
> 

It doesn't make sense to default this behaviour. The BZ suggested to pass 0 as the number of queues passed to libvirt in case there vmi has no interfaces (no matter if multiqueue is specified or not).
Theoretically just removing the error in the bug (blocking running multiqueue vm with no interfaces) and adding this default is problematic. For example, if `AutoattachPodInterface` is false, in case the user would like to hot plug an interface to the VM, the queues would have to be recalculated (I"m not sure if it is supported to hot re-set the queues number).

> This PR suggest something else: to focus and be more accurate in the error thrown, taking into account some other default we have (to add an interface if nothing is specified).
> 

Yes, if we can limit the scope of the error, why not?

> So I guess @dankenigsberg [comment](https://github.com/kubevirt/kubevirt/pull/7432#issuecomment-1076581202) (which I saw no response to) have value and should be considered. Adding logic like this, does complicate the validation, asking the user to fix his definition and know more parameters. If we provide defaults, like defining an interface/network when nothing is specified, it is no less reasonable to adjust the mutique per the input and not complicate life to a simple user.

Reply:
> It doesn't make sense to default this behaviour. The BZ suggested to pass 0 as the number of queues passed to libvirt in case there vmi has no interfaces (no matter if multiqueue is specified or not).
Theoretically just removing the error in the bug (blocking running multiqueue vm with no interfaces) and adding this default is problematic. For example, if AutoattachPodInterface is false, in case the user would like to hot plug an interface to the VM, the queues would have to be recalculated (I"m not sure if it is supported to hot re-set the queues number).

I'm not sure how hotplug got in the way here, or how MQ should behave in that case.
I am sure it just complicates all this even further. (will you blow up with an error when you want to hot-plug the last interface?)

If the template the user used mentions that he wants MQ, but there is no interface that supports it, I see no problem to go on and define it with no MQ.
The same way it makes sense to someone to auto-define an interface to make a user happy, it makes sense to learn the given spec and adjust to a reasonable heuristic that is trivial and covers most of the usages.
Not only that, such logic even simplifies the complex code validation.

I am not going to argue on this strongly, I just think the pros and cons should be placed on the table and discussed. I sympathize with a user not wanting to know what MQ means and still be able to create a VM without interface plugged in.

Reply:
> Aha, it is just hidden behind a wall. Can you please expose it using `libvmi`?

We can't use libvmi instead of `NewRandomVMIWithEphemeralDisk` because the latter depends on test's `checks.IsARM64`